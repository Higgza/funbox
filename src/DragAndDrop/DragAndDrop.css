
.DragAndDrop ul {
    list-style: none;
    padding: 0;
    margin: 0;
    position: relative;
    margin: 20px auto;
}
.DragAndDrop ul li {
    border: 1px solid #dbdbdb;
    margin: 5px auto;
    cursor: move;
    width: 70%;
    position: relative;
    box-shadow: 0 0 1px 1px rgba(0,0,0,.1);
    border-radius: 4px;
    padding: 5px;
}

.DragAndDrop ul li i {
    position: relative;
    font-size: 20px;
    float: right;
    right: 10px;
    cursor: pointer;
    transition: all .3s;
    color: #ff5664;
}
.DragAndDrop ul li i:hover {
    color: red;
}

.testClassName1 {
    border-radius: 4px;
    border: 1px solid #d5d5d5;
    padding: 10px;
    margin: 20px auto;
    max-width: 70%;
}

.testClassName2 {
    border-radius: 4px;
    border: 1px solid #d5d5d5;
    padding: 10px;
    margin: 20px auto;
    max-width: 70%;
}