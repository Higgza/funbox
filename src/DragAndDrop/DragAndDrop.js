import React, {Component} from 'react';
import './DragAndDrop.css';

import {connect} from 'react-redux';
import {cArray, deleteRouteList} from "../redux/actions/MapActions/mapActions";

export class DragAndDrop extends Component {

    state = {
        memory: {}
    };

    DragEnter = (e) => {
        e.target.style.border = '2px dashed #717dff';
    };

    DragLeave = (obj, i, e) => {
        e.target.style.border = '1px solid #dbdbdb';
        this.setState({
            memory: {
                index: i,
                name: obj.name
            }
        });
    };

    changeRouteArray = (obj, i) => {
        let cloneArray = this.props.ddMapArray.concat();
        cloneArray[this.state.memory.index] = obj;
        cloneArray[i] = this.state.memory;

        this.props.onCArray(cloneArray);
    };

    deleteThisAdress = (i) => {
        let ar = this.props.ddMapArray.concat();
        ar.splice(i, 1);
        this.props.onDeteleArrayRoute(ar);
    };

    render() {
        return (
                <div className={"DragAndDrop"}>
                    {   this.props.ddMapArray.length >= 2 ?
                        <div className={'testClassName1'}>
                            <h2>Построенный Вами маршрут</h2>
                            <ul>
                                {
                                    this.props.ddMapArray.map((x, i) => (
                                        <li key={i}
                                            draggable={true}
                                            onDragEnter={this.DragEnter}
                                            onDragLeave={this.DragLeave.bind(this, x, i)}
                                            onDragEnd={this.changeRouteArray.bind(this, x, i)}
                                        >{x.name} <i className="material-icons"
                                                     onClick={this.deleteThisAdress.bind(this, i)}
                                        >delete_forever</i></li>
                                    ))
                                }
                            </ul>
                        </div>
                        : null
                    }
                    { this.props.ddMapArray.length === 1 ?
                            <div className={'testClassName2'}>
                                <h2>Добавте минимум 2 точки маршрута</h2>
                                <ul>
                                    <li>{this.props.ddMapArray[0].name} <i className="material-icons" onClick={this.deleteThisAdress.bind(this, 0)}
                                    >delete_forever</i></li>
                                </ul>
                            </div>
                        : null
                    }
                </div>
        );
    }
}

function s(state) {
    return {
        ddMapArray: state.googleMap.arr
    }
}

function d(dispatch) {
    return {
        onCArray: (obj) => dispatch(cArray(obj)),
        onDeteleArrayRoute: (obj) => dispatch(deleteRouteList(obj))
    }
}

export default connect(s, d)(DragAndDrop);