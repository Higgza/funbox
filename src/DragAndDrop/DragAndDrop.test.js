import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {DragAndDrop} from './DragAndDrop';

configure({
    adapter: new Adapter()
});

describe('Shallow test DragAndDrop.js', () => {

    const props = {
        ddMapArray: [],
        onCArray: jest.fn(),
        onDeteleArrayRoute: jest.fn()
    };

    describe('initial component DragAndDrop', () => {

        const DDComponent = shallow(<DragAndDrop {...props} />);

        it('snapshot test', () => {
            expect(DDComponent).toMatchSnapshot();
        });
    });

    describe('1 length array component DragAndDrop', () => {
        const nextProps = {
            ...props,
            ddMapArray: [
                {name: 'One Name'}
            ]
        };

        const DDComponent = shallow(<DragAndDrop {...nextProps} />);

        it('snapshot test', () => {
            expect(DDComponent).toMatchSnapshot();
        });

        it('list functions test DragAndDrop', () => {
            DDComponent.find('div.testClassName2 > ul > li > i').simulate('click');
            expect(nextProps.onDeteleArrayRoute).toHaveBeenCalled();
        });
    });

    describe('props test component DragAndDrop', () => {
        const nextProps = {
            ...props,
            ddMapArray: [
                {name: 'One Name'},
                {name: 'Two Name'},
                {name: 'Three Name'}
            ]
        };

        const DDComponent = shallow(<DragAndDrop {...nextProps} />);

        it('snapshot test DragAndDrop', () => {
            expect(DDComponent).toMatchSnapshot();
        });

        it('list test DragLeave -- DragAndDrop', () => {
            DDComponent.find('div.testClassName1 > ul > li').first().simulate('DragLeave', {
                target: {
                    style: {
                        border: '1px solid #dbdbdb'
                    }
                }
            });
            expect(DDComponent.state('memory')).toEqual({
                index: 0,
                name: nextProps.ddMapArray[0].name
            });
        });

        it('list test DragEnd -- DragAndDrop', () => {
            DDComponent.find('div.testClassName1 > ul > li').first().simulate('DragEnd');
            expect(props.onCArray).toHaveBeenCalled();
        });
    });
});