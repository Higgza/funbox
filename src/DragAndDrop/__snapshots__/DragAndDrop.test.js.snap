// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow test DragAndDrop.js 1 length array component DragAndDrop snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DragAndDrop
    ddMapArray={
      Array [
        Object {
          "name": "One Name",
        },
      ]
    }
    onCArray={[MockFunction]}
    onDeteleArrayRoute={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <div
          className="testClassName2"
        >
          <h2>
            Добавте минимум 2 точки маршрута
          </h2>
          <ul>
            <li>
              One Name
               
              <i
                className="material-icons"
                onClick={[Function]}
              >
                delete_forever
              </i>
            </li>
          </ul>
        </div>,
      ],
      "className": "DragAndDrop",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Добавте минимум 2 точки маршрута
            </h2>,
            <ul>
              <li>
                One Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>
            </ul>,
          ],
          "className": "testClassName2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Добавте минимум 2 точки маршрута",
            },
            "ref": null,
            "rendered": "Добавте минимум 2 точки маршрута",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li>
                One Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "One Name",
                  " ",
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "One Name",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "delete_forever",
                    "className": "material-icons",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "delete_forever",
                  "type": "i",
                },
              ],
              "type": "li",
            },
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="testClassName2"
          >
            <h2>
              Добавте минимум 2 точки маршрута
            </h2>
            <ul>
              <li>
                One Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>
            </ul>
          </div>,
        ],
        "className": "DragAndDrop",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Добавте минимум 2 точки маршрута
              </h2>,
              <ul>
                <li>
                  One Name
                   
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>
                </li>
              </ul>,
            ],
            "className": "testClassName2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Добавте минимум 2 точки маршрута",
              },
              "ref": null,
              "rendered": "Добавте минимум 2 точки маршрута",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <li>
                  One Name
                   
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>
                </li>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "One Name",
                    " ",
                    <i
                      className="material-icons"
                      onClick={[Function]}
                    >
                      delete_forever
                    </i>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "One Name",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "delete_forever",
                      "className": "material-icons",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "delete_forever",
                    "type": "i",
                  },
                ],
                "type": "li",
              },
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Shallow test DragAndDrop.js initial component DragAndDrop snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DragAndDrop
    ddMapArray={Array []}
    onCArray={[MockFunction]}
    onDeteleArrayRoute={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
      ],
      "className": "DragAndDrop",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
        "className": "DragAndDrop",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Shallow test DragAndDrop.js props test component DragAndDrop snapshot test DragAndDrop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DragAndDrop
    ddMapArray={
      Array [
        Object {
          "name": "One Name",
        },
        Object {
          "name": "Two Name",
        },
        Object {
          "name": "Three Name",
        },
      ]
    }
    onCArray={[MockFunction]}
    onDeteleArrayRoute={
      [MockFunction] {
        "calls": Array [
          Array [
            Array [],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="testClassName1"
        >
          <h2>
            Построенный Вами маршрут
          </h2>
          <ul>
            <li
              draggable={true}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
            >
              One Name
               
              <i
                className="material-icons"
                onClick={[Function]}
              >
                delete_forever
              </i>
            </li>
            <li
              draggable={true}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
            >
              Two Name
               
              <i
                className="material-icons"
                onClick={[Function]}
              >
                delete_forever
              </i>
            </li>
            <li
              draggable={true}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
            >
              Three Name
               
              <i
                className="material-icons"
                onClick={[Function]}
              >
                delete_forever
              </i>
            </li>
          </ul>
        </div>,
        null,
      ],
      "className": "DragAndDrop",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Построенный Вами маршрут
            </h2>,
            <ul>
              <li
                draggable={true}
                onDragEnd={[Function]}
                onDragEnter={[Function]}
                onDragLeave={[Function]}
              >
                One Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>
              <li
                draggable={true}
                onDragEnd={[Function]}
                onDragEnter={[Function]}
                onDragLeave={[Function]}
              >
                Two Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>
              <li
                draggable={true}
                onDragEnd={[Function]}
                onDragEnter={[Function]}
                onDragLeave={[Function]}
              >
                Three Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>
            </ul>,
          ],
          "className": "testClassName1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Построенный Вами маршрут",
            },
            "ref": null,
            "rendered": "Построенный Вами маршрут",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  draggable={true}
                  onDragEnd={[Function]}
                  onDragEnter={[Function]}
                  onDragLeave={[Function]}
                >
                  One Name
                   
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>
                </li>,
                <li
                  draggable={true}
                  onDragEnd={[Function]}
                  onDragEnter={[Function]}
                  onDragLeave={[Function]}
                >
                  Two Name
                   
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>
                </li>,
                <li
                  draggable={true}
                  onDragEnd={[Function]}
                  onDragEnter={[Function]}
                  onDragLeave={[Function]}
                >
                  Three Name
                   
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>
                </li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "One Name",
                    " ",
                    <i
                      className="material-icons"
                      onClick={[Function]}
                    >
                      delete_forever
                    </i>,
                  ],
                  "draggable": true,
                  "onDragEnd": [Function],
                  "onDragEnter": [Function],
                  "onDragLeave": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "One Name",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "delete_forever",
                      "className": "material-icons",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "delete_forever",
                    "type": "i",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Two Name",
                    " ",
                    <i
                      className="material-icons"
                      onClick={[Function]}
                    >
                      delete_forever
                    </i>,
                  ],
                  "draggable": true,
                  "onDragEnd": [Function],
                  "onDragEnter": [Function],
                  "onDragLeave": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Two Name",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "delete_forever",
                      "className": "material-icons",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "delete_forever",
                    "type": "i",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Three Name",
                    " ",
                    <i
                      className="material-icons"
                      onClick={[Function]}
                    >
                      delete_forever
                    </i>,
                  ],
                  "draggable": true,
                  "onDragEnd": [Function],
                  "onDragEnter": [Function],
                  "onDragLeave": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Three Name",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "delete_forever",
                      "className": "material-icons",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "delete_forever",
                    "type": "i",
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="testClassName1"
          >
            <h2>
              Построенный Вами маршрут
            </h2>
            <ul>
              <li
                draggable={true}
                onDragEnd={[Function]}
                onDragEnter={[Function]}
                onDragLeave={[Function]}
              >
                One Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>
              <li
                draggable={true}
                onDragEnd={[Function]}
                onDragEnter={[Function]}
                onDragLeave={[Function]}
              >
                Two Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>
              <li
                draggable={true}
                onDragEnd={[Function]}
                onDragEnter={[Function]}
                onDragLeave={[Function]}
              >
                Three Name
                 
                <i
                  className="material-icons"
                  onClick={[Function]}
                >
                  delete_forever
                </i>
              </li>
            </ul>
          </div>,
          null,
        ],
        "className": "DragAndDrop",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Построенный Вами маршрут
              </h2>,
              <ul>
                <li
                  draggable={true}
                  onDragEnd={[Function]}
                  onDragEnter={[Function]}
                  onDragLeave={[Function]}
                >
                  One Name
                   
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>
                </li>
                <li
                  draggable={true}
                  onDragEnd={[Function]}
                  onDragEnter={[Function]}
                  onDragLeave={[Function]}
                >
                  Two Name
                   
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>
                </li>
                <li
                  draggable={true}
                  onDragEnd={[Function]}
                  onDragEnter={[Function]}
                  onDragLeave={[Function]}
                >
                  Three Name
                   
                  <i
                    className="material-icons"
                    onClick={[Function]}
                  >
                    delete_forever
                  </i>
                </li>
              </ul>,
            ],
            "className": "testClassName1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Построенный Вами маршрут",
              },
              "ref": null,
              "rendered": "Построенный Вами маршрут",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    draggable={true}
                    onDragEnd={[Function]}
                    onDragEnter={[Function]}
                    onDragLeave={[Function]}
                  >
                    One Name
                     
                    <i
                      className="material-icons"
                      onClick={[Function]}
                    >
                      delete_forever
                    </i>
                  </li>,
                  <li
                    draggable={true}
                    onDragEnd={[Function]}
                    onDragEnter={[Function]}
                    onDragLeave={[Function]}
                  >
                    Two Name
                     
                    <i
                      className="material-icons"
                      onClick={[Function]}
                    >
                      delete_forever
                    </i>
                  </li>,
                  <li
                    draggable={true}
                    onDragEnd={[Function]}
                    onDragEnter={[Function]}
                    onDragLeave={[Function]}
                  >
                    Three Name
                     
                    <i
                      className="material-icons"
                      onClick={[Function]}
                    >
                      delete_forever
                    </i>
                  </li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "One Name",
                      " ",
                      <i
                        className="material-icons"
                        onClick={[Function]}
                      >
                        delete_forever
                      </i>,
                    ],
                    "draggable": true,
                    "onDragEnd": [Function],
                    "onDragEnter": [Function],
                    "onDragLeave": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "One Name",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "delete_forever",
                        "className": "material-icons",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "delete_forever",
                      "type": "i",
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Two Name",
                      " ",
                      <i
                        className="material-icons"
                        onClick={[Function]}
                      >
                        delete_forever
                      </i>,
                    ],
                    "draggable": true,
                    "onDragEnd": [Function],
                    "onDragEnter": [Function],
                    "onDragLeave": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Two Name",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "delete_forever",
                        "className": "material-icons",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "delete_forever",
                      "type": "i",
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Three Name",
                      " ",
                      <i
                        className="material-icons"
                        onClick={[Function]}
                      >
                        delete_forever
                      </i>,
                    ],
                    "draggable": true,
                    "onDragEnd": [Function],
                    "onDragEnter": [Function],
                    "onDragLeave": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Three Name",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "delete_forever",
                        "className": "material-icons",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "delete_forever",
                      "type": "i",
                    },
                  ],
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
